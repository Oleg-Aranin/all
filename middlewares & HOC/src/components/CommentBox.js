import React, {Component, useState} from "react";
import {connect} from 'react-redux'
import * as actions from '../actions'
import requireAuth from "./requireAuth";

// const CommentBox = (props) => {
//     const [state, setState] = useState('')
//
//     const handelChange = e => {
//         setState({comment: e.target.value})
//     }
//
//     const handelSubmit = e => {
//         e.preventDefault()
//
//         props.saveComment(state.comment)
//
//         setState({comment: ''})
//     }
//
//
//     return (
//         <div>
//             <form onSubmit={handelSubmit}>
//                 <h4>Add a Comment</h4>
//                 <textarea
//                     onChange={handelChange}
//                     value={state.comment}/>
//                 <div>
//                     <button>Submit Comment</button>
//                 </div>
//             </form>
//             <button className='fetch-comments' onClick={props.fetchComments}>Fetch Comments</button>
//         </div>
//     )
//
// }
//
// function mapStateToProps({auth}) {
//     return {auth}
// }
//
// export default connect(mapStateToProps, actions)(CommentBox)


class CommentBox extends Component {
    state = {comment: ''}


    handelChange = e => {
        this.setState({comment: e.target.value})
    }

    handelSubmit = e => {
        e.preventDefault()

        this.props.saveComment(this.state.comment)

        this.setState({comment: ''})
    }

    render() {
        return (
            <div>
                <form onSubmit={this.handelSubmit}>
                    <h4>Add a Comment</h4>
                    <textarea
                        onChange={this.handelChange}
                        value={this.state.comment}/>
                    <div>
                        <button>Submit Comment</button>
                    </div>
                </form>
                <button className='fetch-comments' onClick={this.props.fetchComments}>Fetch Comments</button>
            </div>
        )
    }
}
export default connect(null, actions)(requireAuth(CommentBox));
