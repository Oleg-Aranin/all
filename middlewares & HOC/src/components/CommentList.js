import React from "react";
import {connect} from "react-redux";

// const CommentList = (props) => {
//     const renderComments = () => {
//         return props.comments.map(comment => {
//             return <li key={comment}>{comment}</li>
//         })
//     }
//
//     return (
//         <div>
//             <ul>
//                 {renderComments()}
//             </ul>
//         </div>
//     )
// }
//
// function mapStateToProps({comments}) {
//     return {comments}
// }
//
// export default connect(mapStateToProps)(CommentList)

class CommentList extends React.Component {

    renderComments() {
        return this.props.comments.map(comment => {
            return <li key={comment}>{comment}</li>
        })
    }

    render() {
        return (
            <div>
                <h4>Comment List</h4>
                <ul>
                    {this.renderComments()}
                </ul>
            </div>
        )
    }
}

function mapStateToProps({comments}) {
    return {comments}
}

export default connect(mapStateToProps)(CommentList)
