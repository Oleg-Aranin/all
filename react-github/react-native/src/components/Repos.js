import React from 'react'
import {View, Text, StyleSheet, FlatList, Linking } from 'react-native'

export default ({repos}) => {
return (
    <FlatList
      data={repos}
      keyExtractor={repos => repos.archive_url}
      renderItem={({item}) => {
    return (
          <View style={styles.container}>
          <Text
              style={styles.name}
              onPress={() => Linking.openURL(item.html_url)}
          > {item.name}</Text>
          </View>
      )
    }}
    />
)
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: 35,

        marginTop: 5,
    },
    name: {
        color: 'blue',
        fontWeight: 'bold',
        alignSelf: 'center',
        fontSize: 16
    }
})
