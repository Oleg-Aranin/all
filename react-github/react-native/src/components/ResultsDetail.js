import React from 'react'
import {View, Text, StyleSheet, Image} from 'react-native'

export default ({result}) => {
    return <View style={styles.container} >
        <Image style={styles.image} source={{uri: result.avatar_url}} />
        <Text style={styles.name} >{result.login}</Text>
    </View>
}

const styles = StyleSheet.create({
    image: {
        width: 115,
        height: 90,
        borderRadius: 4,
        marginBottom: 3,

    },
    name: {
        maxWidth: 100,
        fontWeight: 'bold',

    },
    container: {
        marginLeft: 15,
        marginBottom: 10,

    }
})
