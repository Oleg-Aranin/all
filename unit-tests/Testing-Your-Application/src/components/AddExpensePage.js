import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {addExpense} from '../actions/expenses';

// export class AddExpensePage extends React.Component {
//     onSubmit = (expense) => {
//         this.props.onSubmit(expense);
//         this.props.history.push('/');
//     }
//
//     render() {
//         return (
//             <div>
//                 <h1>Add Expense</h1>
//                 <ExpenseForm
//                     onSubmit={this.onSubmit}
//                 />
//             </div>
//         )
//     }
// }

export const AddExpensePage = (props) => (
    <div>
        <h1>Add Expense</h1>
        <ExpenseForm
            onSubmit={(expense) => {
                props.addExpense(expense);
                props.history.push('/');
            }}
        />
    </div>
);


export default connect(null, {addExpense})(AddExpensePage);
