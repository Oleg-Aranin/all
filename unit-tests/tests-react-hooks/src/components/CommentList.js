import React from "react";
import {connect} from "react-redux";

const CommentList = (props) => {
    const renderComments = () => {
        return props.comments.map(comment => {
            return <li key={comment}>{comment}</li>
        })
    }

    return (
        <div>
            <ul>
                {renderComments()}
            </ul>
        </div>
    )
}

function mapStateToProps({comments}) {
    return {comments}
}

export default connect(mapStateToProps)(CommentList)
//
// class CommentList extends Component {
//
//     renderComments() {
//         return this.props.comments.map(comment => {
//             return <li key={comment}>{comment}</li>
//         })
//     }
//
//     render() {
//         return (
//             <div>
//                 <ul>
//                     {this.renderComments()}
//                 </ul>
//             </div>
//         )
//     }
// }
//
// function mapStateToProps({comments}) {
//     return {comments}
// }
//
// export default connect(mapStateToProps)(CommentList)
