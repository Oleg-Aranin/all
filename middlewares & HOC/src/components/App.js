import React from 'react';
import {Route, Link} from 'react-router-dom'
import {connect} from 'react-redux'
import CommentBox from "./CommentBox";
import CommentList from "./CommentList";
import * as actions from '../actions/index'

class App extends React.Component {
    renderHeader() {
        return (
            <ul>
                <li><Link to='/'>Home</Link></li>
                <li><Link to='/post'>Post A Component</Link></li>
                <li>{this.renderButton()}</li>
            </ul>
        )
    }

    renderButton() {
        if (this.props.auth) {
            return <button onClick={() => this.props.changeAuth(false)}>Sing Out</button>
        } else {
            return <button onClick={() => this.props.changeAuth(true)}>Sing In</button>
        }
    }

    render() {


        return (
            <div>
                {this.renderHeader()}
                <Route path='/post' component={CommentBox}/>
                <Route path='/' exact component={CommentList}/>
            </div>
        );
    }
}

function mapStateToProps({auth}) {
    return {auth}
}

export default connect(mapStateToProps, actions)(App)
