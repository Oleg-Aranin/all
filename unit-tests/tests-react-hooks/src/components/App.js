import React, {useState} from 'react';
import CommentBox from "./CommentBox";
import CommentList from "./CommentList";
import TextInput from "../testing-useEffect/TextInput";

function App() {
    const [value, setValue] = useState('')
    const data = () => {
        return {
            value: 'hh',
            focusedByDefault: true,
            onChange(v) {
                return this.value = v
            }
        }
    }

    return (
        <div>
            <TextInput
                focusedByDefault
                onChange={setValue}
                value={value}
                data={data}
            >
                <p>hi</p>
            </TextInput>

            <CommentBox/>
            <CommentList/>
        </div>
    );
}

export default App
