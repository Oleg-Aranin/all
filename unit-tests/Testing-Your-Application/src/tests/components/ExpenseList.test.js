import React from "react";
import {shallow} from 'enzyme'
import {ExpenseList} from '../../components/ExpenseList'
import expenses from "../fixtures/expenses";
import {MemoryRouter} from "react-router-dom";


describe('ExpenseList', () => {
    it('should render ExpenseList with expenses', () => {
        const wrapper = shallow(
            <MemoryRouter>
                <ExpenseList expenses={expenses}/>
            </MemoryRouter>
        )
        expect(wrapper.html()).toMatchSnapshot()

    })

    it('should render ExpenseList with empty message', () => {
        const wrapper = shallow(
            <MemoryRouter>
                <ExpenseList expenses={[]}/>
            </MemoryRouter>
        )
        expect(wrapper.html()).toMatchSnapshot()

    })

})
