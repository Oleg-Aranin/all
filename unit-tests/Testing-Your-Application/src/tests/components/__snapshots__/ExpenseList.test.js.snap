// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseList should render ExpenseList with empty message 1`] = `"<div><p>No expenses</p></div>"`;

exports[`ExpenseList should render ExpenseList with expenses 1`] = `"<div><div><a href=\\"/edit/1\\"><h3>Gum</h3></a><p>195 - 0</p></div><div><a href=\\"/edit/2\\"><h3>Rent</h3></a><p>109500 - -345600000</p></div><div><a href=\\"/edit/3\\"><h3>Credit Card</h3></a><p>4500 - 345600000</p></div></div>"`;
